File: rust/src/models/subscription_update_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::subscription_details::SubscriptionDetails;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SubscriptionUpdateResponse {
    pub subscription: SubscriptionDetails,
}

impl SubscriptionUpdateResponse {
    pub fn new(subscription: SubscriptionDetails) -> Self {
        Self { subscription }
    }
}

File: typescript/src/models/subscriptionUpdateResponse.ts
// this file is @generated
import {
  type SubscriptionDetails,
  SubscriptionDetailsSerializer,
} from "./subscriptionDetails";

export interface SubscriptionUpdateResponse {
  subscription: SubscriptionDetails;
}

export const SubscriptionUpdateResponseSerializer = {
  _fromJsonObject(object: any): SubscriptionUpdateResponse {
    return {
      subscription: SubscriptionDetailsSerializer._fromJsonObject(object["subscription"]),
    };
  },

  _toJsonObject(self: SubscriptionUpdateResponse): any {
    return {
      subscription: SubscriptionDetailsSerializer._toJsonObject(self.subscription),
    };
  },
};
